src/compiler/checker.ts

```diff
@@ -6867,20 +6867,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                 }
                 else {
-                    const isInstantiationExpressionType = !!(getObjectFlags(type) & ObjectFlags.InstantiationExpressionType);
-                    if (isInstantiationExpressionType) {
-                        const instantiationExpressionType = type as InstantiationExpressionType;
-                        if (isTypeQueryNode(instantiationExpressionType.node)) {
-                            const typeNode = serializeExistingTypeNode(context, instantiationExpressionType.node);
-                            if (typeNode) {
-                                return typeNode;
-                            }
-                        }
-                        if (context.visitedTypes?.has(typeId)) {
-                            return createElidedInformationPlaceholder(context);
-                        }
-                        return visitAndTransformType(type, createTypeNodeFromObjectType);
-                    }
                     // Anonymous types without a symbol are never circular.

```

Comment:
If I understood this correctly, now that anonymous types that are instantiated always have a symbol, then the invariant stated in this comment holds again and we can get rid of the special code here, right?

And the invariant that "Anonymous types without a symbol are never circular" comes from the fact that you couldn't instantiate something that didn't have a name/declaration and therefore a symbol before instantiation expressions existed?

Reply:
That's my understanding, yeah.